File: java/src/main/java/com/mokelab/demo/notification/MainFragment.java
package com.mokelab.demo.notification;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.ListFragment;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.ListView;

/**
 * MainFragment
 */

public class MainFragment extends ListFragment {
    public static MainFragment newInstance() {
        return new MainFragment();
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        String[] labels = getResources().getStringArray(R.array.main_menu);
        ArrayAdapter<String> adapter = new ArrayAdapter<>(getActivity(), android.R.layout.simple_list_item_1, labels);
        setListAdapter(adapter);
    }

    @Override
    public void onListItemClick(ListView l, View v, int position, long id) {
        super.onListItemClick(l, v, position, id);
        switch (position) {
        case 0:
            FragmentUtils.toNextFragment(getFragmentManager(), R.id.container,
                    ChannelFragment.newInstance(), true);
            break;
        case 1:
            FragmentUtils.toNextFragment(getFragmentManager(), R.id.container,
                    ShowNotificationFragment.newInstance(), true);
            break;
        }
    }
}

File: app/src/main/java/com/mokelab/demo/notification/MainFragment.kt
package com.mokelab.demo.notification

import android.os.Bundle
import android.support.v4.app.ListFragment
import android.view.View
import android.widget.ArrayAdapter
import android.widget.ListView

/**
 * MainFragment
 */
class MainFragment : ListFragment() {

    companion object Factory {
        fun newInstance() = MainFragment()
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        val labels = resources.getStringArray(R.array.main_menu)
        this.listAdapter = ArrayAdapter<String>(activity, android.R.layout.simple_list_item_1, labels)
    }

    override fun onListItemClick(l: ListView?, v: View?, position: Int, id: Long) {
        super.onListItemClick(l, v, position, id)
        FragmentUtils.toNextFragment(fragmentManager, R.id.container, when(position) {
            0 -> ChannelFragment.newInstance()
            1 -> ShowNotificationFragment.newInstance()
            else -> ChannelFragment.newInstance()
        }, true)
    }
}